sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs

```diff
@@ -53,6 +53,14 @@ private static void InitializeStatsbeat(ConnectionVars connectionVars)
                 // Do not initialize statsbeat for statsbeat.
                 if (connectionVars != null && connectionVars.InstrumentationKey != ConnectionStringParser.GetValues(Statsbeat.Statsbeat_ConnectionString_EU).InstrumentationKey && connectionVars.InstrumentationKey != ConnectionStringParser.GetValues(Statsbeat.Statsbeat_ConnectionString_NonEU).InstrumentationKey)
                 {
+                    var disableStatsbeat = Environment.GetEnvironmentVariable("APPLICATIONINSIGHTS_STATSBEAT_DISABLED");
```

Comment:
Should we check this logic in constructor, before calling `InitializeStatsbeat`?

Reply:
We will run it atleast twice in that case, one for user's ikey and one for statsbeat itself. Putting it here prevents checking for statsbeat itself.